montepy/utilities.py

```diff
@@ -31,6 +32,27 @@ def fortran_float(number_string):
             raise ValueError(f"Value Not parsable as float: {number_string}") from e
 
 
+def is_comment(line):
+    """
+    Determines if the line is a ``C comment`` style comment.
+
+    :param line: the line to analyze
+    :type line: str
+    :returns: True if the line is a comment
+    :rtype: bool
+    """
+    upper_start = line[0 : BLANK_SPACE_CONTINUE + 1].upper()
+    non_blank_comment = upper_start and line.lstrip().upper().startswith("C ")
+    if non_blank_comment:
+        return True
+    blank_comment = (
+        "C\n" == upper_start.lstrip()
+        or "C\r\n" == upper_start.lstrip()
+        or ("C" == upper_start and "\n" not in line)
```

Comment:
Why not just `"C".strip()`?

Reply:
I'd have to dig through the git history, but there were some edge cases that didn't work for this IIRC. It was really annoying and this isn't clean, but it works for all edge cases so far.

Reply:
If you can find one of the edge cases that breaks `"C".strip()`, think about discussing that in the function docstring or a code comment.

Reply:
Ok I think you were right. Is this new diff what you were thinking?

Reply:
Yeah that's better